client/src/pages/Explorer.js

```diff
@@ -73,47 +98,82 @@ function FavoriteCheckBox() {
+        });
+      } catch (error) {
+         console.log();
+      }
+    }
+    getData();
+  }, [userFavoriteCitiesUrl, citiesUrl]);
   return (
-		<Container className={classes.pageContainer}>
-			<Typography variant="h4" className={classes.title}>
-				Explore destinations
-			</Typography>
-			<Typography
-				variant="subtitle2"
-				className={classes.title}
-				style={{ color: "#c5bec4", fontSize: 12 }}
-			>
-				World's top destinations to explore
-			</Typography>
-			<Grid container spacing={3} justify="center" style={{ marginTop: 32, height: "75%" }}>
-				{places.map((place) => (
-					<Grid item key={place.countryName} xs={12} sm={3}>
-						<div
-							className={classes.paperContainer}
-							style={{
-								backgroundImage: `linear-gradient(to bottom, transparent, rgba(52, 52, 52, 0.63)), url(${place.imageUrl})`,
-								// height: "100%",
-							}}
-						>
-							<div className={classes.bottomInformationContainer}>
-								<span className={classes.bottomInformationSubContainer1}>
-									<span style={{ fontSize: 17, color: "white" }}>
-										{place.townName},
-									</span>
-									<span style={{ fontSize: 11, color: "rgb(175 175 175)" }}>
-										{place.countryName}
-									</span>
-								</span>
-								<span className={classes.bottomInformationSubContainer2}>
-									<FavoriteCheckBox />
-								</span>
-							</div>
-						</div>
-					</Grid>
-				))}
-			</Grid>
-		</Container>
-	);
+    <Container className={classes.pageContainer}>
+      <Typography variant="h4" className={classes.title}>
+        Explore destinations
+      </Typography>
+      <Typography
+        variant="subtitle2"
+        className={classes.title}
+        style={{ color: "#c5bec4", fontSize: 12 }}
+      >
+        World's top destinations to explore
+      </Typography>
+      <Grid
+        container
+        spacing={2}
+        justify="center"
+        style={{ maxWidth: "75%", marginTop: 32 }}
+      >
+        {generalData.places.map((place) => (
+          <Grid item key={place.name}>
+            <div
+              className={classes.paperContainer}
+              style={{
+                backgroundImage: `linear-gradient(to bottom, transparent, rgba(52, 52, 52, 0.63)), url(${place.imageUrl})`,
+              }}
+            >
+              <div className={classes.bottomInformationContainer}>
+                <span className={classes.bottomInformationSubContainer1}>
+                  <span style={{ fontSize: 17, color: "white" }}>
+                    {place.name},
+                  </span>
+                  <span style={{ fontSize: 11, color: "rgb(175 175 175)" }}>
+                    {place.country}
+                  </span>
+                </span>
+                <span className={classes.bottomInformationSubContainer2}>
+                  <FavoriteCheckBox currentPlace={place} favorite={generalData.favorites.indexOf(place.name) >=0 }/>
```

Comment:
instead of doing this nested loop, we can just add a "liked" or "favorited" boolean to each place object in state